meson.build

```diff
@@ -53,6 +53,9 @@ endif
 if not cy.version().version_compare('>=0.29.35')
   error('SciPy requires Cython >= 0.29.35')
 endif
+if not cy.version().version_compare('<3.0')
+  error('SciPy requires Cython < 3.0')
+endif
```

Comment:
This doesn't actually work exactly as intended unfortunately, because `.version_compare` does not deal with pre-releases. So a `3.0.0b3` version will choke on this check. Now, that may be intended since those pre-releases aren't really supported. However, the 3.0 release is expected soon and it actually does work. So I think we should remove this check again. We'll keep it in `pyproject.toml`, so by default users who `pip install scipy` and get a from-source build will get the latest 0.29.x version. But if they are explicitly using 3.0, let's let them do that.